geowebcache/core/src/main/java/org/geowebcache/storage/blobstore/memory/MemoryBlobStore.java

```diff
@@ -742,4 +757,51 @@ public boolean executeOperation(BlobStore store, Object... objs)
         public abstract boolean executeOperation(BlobStore store, Object... objs)
                 throws StorageException;
     }
+
+    @Override
+    public boolean deleteByParametersId(String layerName, String parametersId)
+            throws StorageException {
+        componentsStateLock.lock();
+        try {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Removing Layer: " + layerName);
+            }
+            // Remove the layer from the cacheProvider
+            cacheProvider.removeLayer(layerName);
```

Comment:
Removes the entire layer? I guess it would have been too hard to remove just the params, but could you elaborate? Maybe in the code comment itself?